dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -101,8 +101,8 @@ def new_general_foorm(survey_names:, day:, subject: FOORM_GENERAL_SURVEY_SUBJECT
     # GET /pd/workshop_survey/post/:enrollment_code
     # If Build Your Own, redirect to its specific survey link. Otherwise, use Foorm.
     def new_post
-      course = get_workshop_by_enrollment_or_course_and_subject(enrollment_code: params[:enrollment_code], course: nil, subject: nil)&.course
-      if course == COURSE_BUILD_YOUR_OWN
+      workshop = get_workshop_by_enrollment_or_course_and_subject(enrollment_code: params[:enrollment_code], course: nil, subject: nil)
```

Comment:
if this is `false`, what should happen? Should the else branch of the if/else below still happen?

Reply:
Yep!